connectors/sources/graphql.py

```diff
@@ -0,0 +1,512 @@
+                "type": "list",
+            },
+            "headers": {
+                "label": "Headers",
+                "order": 10,
+                "type": "str",
+                "required": False,
+            },
+            "pagination_model": {
+                "display": "dropdown",
+                "label": "Pagination model",
+                "options": [
+                    {"label": "No pagination", "value": "no_pagination"},
+                    {"label": "Cursor-based pagination", "value": "cursor_pagination"},
+                ],
+                "order": 11,
+                "tooltip": "Cursor based pagination requires 'pageInfo' field along with argument 'after' for objects mentioned in 'GraphQL Objects List'. It also requires variable for 'after' argument.",
+                "type": "str",
+                "value": "none",
+            },
+            "pagination_key": {
+                "depends_on": [
+                    {"field": "pagination_model", "value": "cursor_pagination"}
+                ],
+                "label": "Pagination key",
+                "order": 12,
+                "tooltip": "Specifies which GraphQL object is used for pagination. Use '.' to provide full path of the object from the root of the response. For example 'organization.users'",
+                "type": "str",
+            },
+            "connection_timeout": {
+                "default_value": 300,
+                "display": "numeric",
+                "label": "Connection Timeout",
+                "order": 13,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+            },
+        }
+
+    def is_query(self, graphql_query):
+        try:
+            ast = parse(graphql_query)  # pyright: ignore
+            for definition in ast.definitions:  # pyright: ignore
+                if (
+                    hasattr(definition, "operation")
+                    and definition.operation.value != "query"
+                ):
+                    return False
+            return True
+        except Exception as e:
+            self._logger.error(f"Failed to parse GraphQL query. Error: {e}")
+            return False
+
+    def validate_endpoints(self):
+        if re.match(URL_REGEX, self.graphql_client.url):
+            return True
+        return False
+
+    async def validate_config(self):
+        """Validates whether user input is empty or not for configuration fields
+        Also validate, if user configured repositories are accessible or not and scope of the token
+        """
+        await super().validate_config()
+
+        if not self.validate_endpoints():
+            msg = "HTTP URL & Endpoint are not structured."
+            raise ConfigurableFieldValueError(msg)
+
+        if not self.is_query(graphql_query=self.graphql_client.graphql_query):
+            msg = "Configured Query is not supported by the connector."
```

Comment:
Can you elaborate that the query is either invalid or is not a query e.g. it's mutation?